### PR TITLE
chore: Add soak analysis as PR comment

### LINKED ISSUE #9622: On-demand trigger of remote soak for Vector PR with telemetry capture, PR feedback with extant and PR introduced soaks

### PR DESCRIPTION
**Pull request by blt**
This commit adds an additional step to the soak analysis which comments on
pull-requests as soaks wrap up. This makes it easier for folks to see if
their change adjusts vector performance, relative to the baseline.

Closes #9622

Signed-off-by: Brian L. Troutwine <brian@troutwine.us>

### CODE REVIEW

**Comment by blt** [comment]
Once this works we'll see a comment dropped on this PR.

**Comment by jszwedko** [comment]
Looking good! I'm wondering if it'd make sense to just have a single table that looks like:

| TEST | EXPERIMENT | VALUE_min | VALUE_p90 | VALUE_p99 | VALUE_max | VALUE_skewness | VALUE_kurtosis |
| --- | --- | --- | --- | --- | --- | --- | --- |
| syslog_regex_logs2metric_ddmetrics | baseline | 16.27Mi | 16.38Mi | 16.40Mi | 16.40Mi | -0.16 | -0.16 |
| syslog_regex_logs2metric_ddmetrics | comparison | 16.72Mi | 16.86Mi | 16.89Mi | 16.89Mi | -0.09 | -0.82 |
| syslog_splunk_hec_logs| baseline | 28.75Mi | 28.82Mi | 28.84Mi | 28.85Mi | -0.14 | -0.50 |
| syslog_splunk_hec_logs| comparison | 28.64Mi | 28.74Mi | 28.75Mi | 28.75Mi | 0.17 | -1.13 |

To make it a bit more concise.

**Comment by spencergilbert** [comment]
I'd definitely lean the expandable sections as suggested by @tobz

File: .github/workflows/soak.yml [hunk 1]
```diff
@@ -245,40 +265,17 @@ jobs:
...
-          context: .
-          file: lib/soak/Dockerfile
-          push: true
-          tags: ${{ steps.meta.outputs.tags }}
-          labels: ${{ steps.meta.outputs.labels }}
-          cache-from: type=gha, scope=${{ github.workflow }}
-          cache-to: type=gha, scope=${{ github.workflow }}
+          issue-number: ${{ needs.compute-soak-meta.outputs.pr-number }}
+          edit-mode: replace
+          body: ${{ steps.read-analysis.outputs.content }}
```
Comment (jszwedko) [inline .github/workflows/soak.yml:281]: These actions seam simple enough that I think I would have preferred to just use a script action than pull in extra dependencies but it is not a huge deal.

**Comment by jszwedko** [comment]
> I'd definitely lean the expandable sections as suggested by @tobz

My worry with expandable sections is that no one will ever expand them 😄  If anything, I'd make the whole soak test result expandable so that it could be expanded, scanned, and then collapsed.

This could also be resolved by having additional user feedback aside from just the raw results (e.g. +/- %s, flagging regressions, etc.) that would prompt expanding.

File: .github/workflows/soak.yml (hunk 1 above)
Comment (blt) [reply .github/workflows/soak.yml:281]: You would think, but I can't write javascript. I'm open to PRs that remove the need for these.

**Comment by blt** [comment]
@tobz @jszwedko @spencergilbert @fuchsnj concerning the suggestions for how to adjust the comment structure, can you drop your comments over in https://github.com/vectordotdev/vector/issues/9926? It's something we need to play with for sure but I don't want to pollute this PR with more changes if it can be helped.

**Pull request merged**